test/test_sparse_csr.py

```diff
@@ -134,6 +134,83 @@ def test_empty_errors(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, "torch.empty: Only 2D sparse CSR tensors are supported."):
             torch.empty((2, 3, 4), dtype=dtype, device=device, layout=torch.sparse_csr)
 
+    @skipMeta
+    @dtypes(*get_all_dtypes())
+    def test_copy(self, device, dtype):
+
+        def run_test(shape, nnz, index_type):
+            a = self.genSparseCSRTensor(shape, nnz, dtype=dtype, device=device, index_dtype=index_dtype)
+            b = self.genSparseCSRTensor(shape, nnz, dtype=dtype, device=device, index_dtype=index_dtype)
+
+            a.copy_(b)
+
+            self.assertEqual(a.crow_indices(), b.crow_indices())
+            self.assertEqual(a.col_indices(), b.col_indices())
+            self.assertEqual(a.values(), b.values())
+
+        ns = [5, 2, 0]
+        for shape, index_dtype in zip(itertools.product(ns, ns), [torch.int32, torch.int64]):
+            run_test(shape, 0, index_dtype)
+            run_test(shape, shape[0]*shape[1], index_dtype)
+
+    @skipMeta
+    @dtypes(*get_all_dtypes())
+    def test_copy_errors(self, device, dtype):
+        for index_dtype in [torch.int32, torch.int64]:
+            shape1 = (2, 3)
+            shape2 = (3, 2)
+            a = self.genSparseCSRTensor(shape1, 0, dtype=dtype, device=device, index_dtype=index_dtype)
+            b = self.genSparseCSRTensor(shape2, 0, dtype=dtype, device=device, index_dtype=index_dtype)
+
+            with self.assertRaisesRegex(RuntimeError, "only same size tensors are supported."):
+                a.copy_(b)
+
+            with self.assertRaisesRegex(RuntimeError, "copy between different layouts is not supported."):
+                a.copy_(torch.empty(a.shape, dtype=dtype, device=device))
+
+            b = self.genSparseCSRTensor(shape1, 1, dtype=dtype, device=device, index_dtype=index_dtype)
+            with self.assertRaisesRegex(RuntimeError, "only tensors with the same number of specified elements are supported."):
+                a.copy_(b)
+
+    @skipMeta
+    @dtypes(*get_all_dtypes())
+    def test_resize(self, device, dtype):
```

Comment:
One feature of resize is that if the [memory shrinks the values are preserved](https://pytorch.org/docs/master/generated/torch.Tensor.resize_.html?highlight=resize). I think this should be true as well here since we use resize_ to change values, but it could be good to add a test just to make sure and in particular when resizing each dimension separately.

Reply:
The behavior is similar here, yes because of `resize_` on strided tensors, but only for column indices and values. The last element of row indices is modified with the resize operation. Currently resizing to a smaller size and then back to the original size would not result in the same original matrix.

Reply:
Added another comment higher up in the code. What do you mean by ```the last element of row indices is modified with the resize operation```?